### PR TITLE
[IMP] tests, web: improve test_js end catching

### PR DESCRIPTION
**Pull request by Xavier-Do**
Since #99912 logging an error message doesn't always end qunit tests. This was mainly to allow to failfast logging qunit errors earlier without stopping the tests in order to test all qunit anyway.

The logic was to have an end message that stops the test.

Unfortunately some errors will prevent the qunit suite to start and the test will wait a 1800 long timer. An example was because of a Missing dependencies. https://runbot.odoo.com/runbot/build/19306352

This new approach will avoid to stop only if the message looks like a qunit failure and the final message is not there (to be sure).

### CODE REVIEW

File: addons/web/tests/test_js.py [hunk 1]
```diff
@@ -45,4 +59,4 @@ class MobileWebSuite(odoo.tests.HttpCase):
 
     def test_mobile_js(self):
         # webclient mobile test suite
-        self.browser_js('/web/tests/mobile?mod=web', "", "", login='admin', timeout=1800, failure_message=' tests failed.')
+        self.browser_js('/web/tests/mobile?mod=web', "", "", login='admin', timeout=1800, failure_message=' tests failed.', error_checker=qunit_error_checker)
```
Comment (d-fence) [inline addons/web/tests/test_js.py:62]: `failure_message` left over.
```suggestion
        self.browser_js('/web/tests/mobile?mod=web', "", "", login='admin', timeout=1800,  error_checker=qunit_error_checker)
```

**Review by d-fence — CHANGES_REQUESTED** [review]
Nice.

**Comment by Xavier-Do** [comment]
@robodoo r+

**Comment by Xavier-Do** [comment]
@rebase-ff

**Comment by ryv-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**